bacnet/src/whois.rs

```diff
@@ -25,7 +25,7 @@ pub struct IAmDevice {
     pub vendor_id: u16,
     pub mac_addr: [u8; 6],
     pub network_number: u16,
-    pub addr: [u8; 6],
+    pub addr: Vec<u8>,
```

Comment:
Is there a reason for preferring a Vec<u8>  over a fixed array of length 6?

Reply:
According to the standard `addr` can have a variable length, from 1 to 6 depending on the data link/physical layer used. That's why I used Vec<u8>.

Reply:
Yes, but in both the places that I see it being used, it's initialized to a vector of six elements, so it's not really leveraging the variable length.

Reply:
I'm wondering if we can't get away with something simpler/better than a generic Vec here... I'll think about it.

Reply:
Can even change the last bit to:

```rust
        let dadr = dadr.as_ref();
        addr.len = dadr.len() as u8;
        addr.adr[..dadr.len()].copy_from_slice(dadr);
```